File: solutions/2026-01-03/solution.py
# Container With Most Water
# Difficulty: Hard
# URL: https://leetcode.com/problems/container-with-most-water/

def maxArea(height):
  max_water = 0
  left = 0
  right = len(height) - 1
  while left < right:
    width = right - left
    area = width * min(height[left], height[right])
    max_water = max(max_water, area)
    if height[left] < height[right]:
      left += 1
    else:
      right -= 1
  return max_water

File: solutions/2026-01-03/solution.java
// Container With Most Water
// Difficulty: Hard
// URL: https://leetcode.com/problems/container-with-most-water/

public int maxArea(int[] height) {
  int max_water = 0;
  int left = 0;
  int right = height.length - 1;
  while (left < right) {
    int width = right - left;
    int area = width * Math.min(height[left], height[right]);
    max_water = Math.max(max_water, area);
    if (height[left] < height[right]) {
      left++;
    } else {
      right--;
    }
  }
  return max_water;
}